ext/sodium/libsodium.c

```diff
@@ -0,0 +1,2682 @@
+		return;
+	}
+	if (opslimit < crypto_pwhash_OPSLIMIT_INTERACTIVE) {
+		zend_error(E_WARNING,
+				   "number of operations for the argon2i function is low");
+	}
+	if (memlimit < crypto_pwhash_MEMLIMIT_INTERACTIVE) {
+		zend_error(E_WARNING, "maximum memory for the argon2i function is low");
+	}
+	hash = zend_string_alloc((size_t) hash_len, 0);
+	if (crypto_pwhash
+		((unsigned char *) ZSTR_VAL(hash), (unsigned long long) hash_len,
+		 passwd, (unsigned long long) passwd_len, salt,
+		 (unsigned long long) opslimit, (size_t) memlimit,
+		 crypto_pwhash_alg_default()) != 0) {
+		zend_string_free(hash);
+		zend_throw_exception(sodium_exception_ce, "crypto_pwhash(): internal error", 0);
+		return;
+	}
+	ZSTR_VAL(hash)[hash_len] = 0;
+
+	RETURN_STR(hash);
+}
+
+PHP_FUNCTION(sodium_crypto_pwhash_str)
+{
+	zend_string *hash_str;
+	char		*passwd;
+	zend_long	 memlimit;
+	zend_long	 opslimit;
+	size_t		 passwd_len;
+	size_t		 len;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "sll",
+							  &passwd, &passwd_len,
+							  &opslimit, &memlimit) == FAILURE ||
+		opslimit <= 0 || memlimit <= 0 || memlimit > SIZE_MAX) {
+		zend_throw_exception(sodium_exception_ce,
+				   "crypto_pwhash_str(): invalid parameters",
+				   0);
+		return;
+	}
+	if (passwd_len <= 0) {
+		zend_error(E_WARNING, "empty password");
+	}
+	if (opslimit < crypto_pwhash_OPSLIMIT_INTERACTIVE) {
+		zend_error(E_WARNING,
+				   "number of operations for the argon2i function is low");
+	}
+	if (memlimit < crypto_pwhash_MEMLIMIT_INTERACTIVE) {
+		zend_error(E_WARNING,
+				   "maximum memory for the argon2i function is low");
+	}
+	hash_str = zend_string_alloc(crypto_pwhash_STRBYTES - 1, 0);
+	if (crypto_pwhash_str
+		(ZSTR_VAL(hash_str), passwd, (unsigned long long) passwd_len,
+		 (unsigned long long) opslimit, (size_t) memlimit) != 0) {
+		zend_string_free(hash_str);
+		zend_throw_exception(sodium_exception_ce, "crypto_pwhash_str(): internal error", 0);
+		return;
+	}
+	ZSTR_VAL(hash_str)[crypto_pwhash_STRBYTES - 1] = 0;
+
+	len = strlen(ZSTR_VAL(hash_str));
+	ZSTR_TRUNCATE(hash_str, len);
+
+	RETURN_STR(hash_str);
+}
+
+PHP_FUNCTION(sodium_crypto_pwhash_str_verify)
+{
+	char	  *hash_str;
+	char	  *passwd;
+	size_t	   hash_str_len;
+	size_t	   passwd_len;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "ss",
+							  &hash_str, &hash_str_len,
+							  &passwd, &passwd_len) == FAILURE) {
+		zend_throw_exception(sodium_exception_ce,
+				   "crypto_pwhash_str_verify(): invalid parameters",
+				   0);
+		return;
+	}
+	if (passwd_len <= 0) {
+		zend_error(E_WARNING, "empty password");
+	}
+	if (crypto_pwhash_str_verify
+		(hash_str, passwd, (unsigned long long) passwd_len) == 0) {
+		RETURN_TRUE;
+	}
+	RETURN_FALSE;
+}
+#endif
+
+PHP_FUNCTION(sodium_crypto_aead_aes256gcm_is_available)
+{
+#ifdef HAVE_AESGCM
+	RETURN_BOOL(crypto_aead_aes256gcm_is_available());
+#else
+	RETURN_FALSE;
+#endif
+}
```

Comment:
@jedisct1 meaning the functions are not defined on those systems. Imho compilation of the extension on such old systems must not be supported, This is all about security after all and those systems are not secure anymore anyways. 😉 

@paragonie-scott seems dangerous right now, it seems as if 1.0.11 is the one available on current distros. However, this should be discussed with all of internals, I cannot decide that.